optuna/terminator/callback.py

```diff
@@ -66,7 +64,7 @@ def objective(trial):
 
     def __init__(
         self,
-        terminator: Optional[BaseTerminator] = None,
+        terminator: BaseTerminator | None = None,
     ) -> None:
```

Comment:
```suggestion
    def __init__(self, terminator: BaseTerminator | None = None):
```

Reply:
I changed the code except that I left `-> None`.